templates/_worker.js/index.ts

```diff
@@ -147,9 +149,19 @@ export default {
       for (const matcher of matchers) {
         if (matcher.regexp) {
           const regexp = new RegExp(matcher?.regexp);
+          const nextPathname = __BASE_PATH__
+            ? // Remove basePath from URL, also squish `//` into `/`
+              // If the baseUrl is set to "/docs" the following will happen:
+              // `/docs/profile/settings` -> `/profile/settings`
+              // `/docs` -> `/`
+              // `/docs/` -> `/`
+              // `/docs/_next/static/main.js` -> `/_next/static/main.js`
+              pathname.replace(__BASE_PATH__, "/").replace("//", "/")
+            : pathname;
+
           if (
-            pathname.match(regexp) ||
-            `${pathname}/page`.replace("//page", "/page").match(regexp)
+            nextPathname.match(regexp) ||
+            `${nextPathname}/page`.replace("//page", "/page").match(regexp)
```

Comment:
is this replace still needed since you're squashing `//` into `/` already?

Reply:
I think this is still needed in the case that bathPath isn't used

Reply:
ah yeah sorry, it is, my bad 😅